packages/l2b/src/implementations/discovery-ui/getCode.ts

```diff
@@ -45,6 +74,17 @@ export function getCode(
     address,
   )
 
+  const flatCodeCurrent = isFlatCodeCurrent(
+    configReader,
+    project,
+    address,
+    codePaths,
+  )
+
+  if (checkFlatCode && !flatCodeCurrent) {
+    throw new Error('Flat code is outdated')
+  }
+
```

Comment:
Nit: we can spare some reads here - won't make a huge difference tho

```suggestion
  if (checkFlatCode) {
    if (!isFlatCodeCurrent(configReader, project, address, codePaths)) {
      throw new Error('Flat code is outdated')
    }
  }
```